### DISCUSSION TITLE
TypeHero Direction

### DISCUSSION THREAD

**Discussion by bautistaaa** [typehero/typehero#1110] [General]
For those of you interested in the future of TypeHero, here's a comprehensive overview of the key engineering areas we're prioritizing.

# Typehero 2.0 Roadmap

## Social Platform Features

For a thriving community, it's essential to provide members with real-time communication features. The following features outline the minimum viable product (MVP) for the social aspect of the platform.

### **Notifications**

Members currently lack the ability to receive in-app or email notifications. This feature is crucial for community vitality, as it prevents interactions from becoming isolated. Notifications also play a vital role in directing traffic back to the platform.

### Tagging

Users should have the capability to tag each other, facilitating ongoing discussions and initiating conversations at any point on the platform. This not only boosts traffic but also enhances discoverability.

## Typescript Playground

The current Typescript playground offers a basic experience. We aim to elevate this experience by introducing additional features, making it the preferred Typescript playground. Recognizing that engineers may not want to engage in Typescript challenges constantly, we aspire to create a sandbox akin to CodePen. This will enable users to test Typescript features, debug issues, and share ideas, resulting in a consistent flow of traffic, similar to platforms like CodePen, CodeSandbox, and the TS Playground.

### Benchmarking

Introduce the capability to benchmark Typescript code(run tests on the server?). This innovative feature sets our playground apart, as benchmarking is currently under explored in the Typescript space. Furthermore, we can leverage benchmarking to rank solutions and present challenges, similar to the approach taken by platforms like LeetCode.

## **Improving User Profiles**

Extend the functionality of user profiles based on community requests.

## **Expanding Content**

Enhance challenge creation and further build out tracks. Tracks are considered crucial for attracting individuals aiming to improve their TypeScript skills. Continue to build out the pipeline around this area to test existing solutions and ensure quality(linting/schema validation, etc)

## Moderation
Continue to build out the admin experience to enable moderators to effectively handle issues in the community. Further, enhance the user experience of users that report issues to the team whether through RTC or other means.

## **Miscellaneous**

Implement a spotlight search for challenges to boost discoverability.

Introduce the ability to tag challenges for improved recommendations and discoverability.

Investigate methods to incorporate seasonal challenges or events for the community.

Increase overall stability of the code editor.

**Comment by bautistaaa** [comment]
Stretch goal: Build out the Advent of Typescript platform

**Comment by jhechtf** [comment]
I definitely like the idea of the TS benchmarking. I know Arktype released something about their initial work into benchmarking so that will be interesting to get going. Do we want to make a more formal roadmap? the dates obviously would be much more flexible since this is OSS but it would be a nice thing (for me) so that I could remember to prioritize some of my TypeHero work since i get distracted.

**Reply by bautistaaa** [reply]
i think less dates and just boards(epics) around each initiative.
